### PR TITLE
feat(sparksql): Add Spark-specific sequence function for integer types

### PR DESCRIPTION
**Pull request by zjuwangg**
Add Spark `sequence` function support for all integer types (tinyint, smallint, integer, bigint) by extracting the shared `SequenceFunction<T, K>` template into `bolt/functions/lib/` for reuse across dialects.

- Extract `SequenceFunction` template into `lib/Sequence.h` (declaration) and `lib/Sequence.cpp` for reuse across dialects.
- Add `sparksql/SequenceFunction.cpp` with Spark signatures and explicit instantiations for `int8_t` and `int16_t` types.
- Keep Presto-specific signatures and registration in `prestosql/Sequence.cpp`.
- Add Spark-specific tests covering all integer types, null handling, and error cases.

Reference:
https://github.com/facebookincubator/velox/commit/23b491ec0e4df2bcb9b93fcfc7378b0e176985d0

### What problem does this PR solve?

Issue Number: close #xxx

### Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🚀 Performance improvement (optimization)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔨 Refactoring (no logic changes)
- [ ] 🔧 Build/CI or Infrastructure changes
- [ ] 📝 Documentation only

### Description

Describe your changes in detail.
For complex logic, explain the "Why" and "How".

### Performance Impact

- [ ✅ ] **No Impact**: This change does not affect the critical path (e.g., build system, doc, error handling).
- [ ] **Positive Impact**: I have run benchmarks.
    <details>
    <summary>Click to view Benchmark Results</summary>

    ```text
    Paste your google-benchmark or TPC-H results here.
    Before: 10.5s
    After:   8.2s  (+20%)
    ```
    </details>
- [ ] **Negative Impact**: Explained below (e.g., trade-off for correctness).

### Release Note

Please describe the changes in this PR

Release Note:

```text
Release Note:
- Fixed a crash in `substr` when input is null.
- optimized `group by` performance by 20%.
```

### Checklist (For Author)

- [ ✅] I have added/updated unit tests (ctest).
- [ ] I have verified the code with local build (Release/Debug).
- [ ] I have run clang-format / linters.
- [ ] (Optional) I have run Sanitizers (ASAN/TSAN) locally for complex C++ changes.
- [ ] No need to test or manual test.

### Breaking Changes

- [✅ ] No
- [ ] Yes (Description: ...)
    <details>
    <summary>Click to view Breaking Changes</summary>

    ```text
    Breaking Changes:
    - Description of the breaking change.
    - Possible solutions or workarounds.
    - Any other relevant information.
    ```
    </details>

### CODE REVIEW

**Comment by zjuwangg** [comment]
@guhaiyan0221 @taiyang-li please help review this merge request when you  guys have free time.

File: bolt/functions/lib/Sequence.cpp [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+  return step / kMillisInDay;
+}
+
+// Explicit instantiations for all supported types.
+// Presto types:
+template class SequenceFunction<int64_t, int64_t>;
+template class SequenceFunction<int32_t, int64_t>;
+template class SequenceFunction<int32_t, int32_t>;
+template class SequenceFunction<Timestamp, int64_t>;
+template class SequenceFunction<Timestamp, int32_t>;
```
Comment (zhangxffff) [inline bolt/functions/lib/Sequence.cpp:283]: Should we support Timestamp in spark Sequence?

Comment (zjuwangg) [reply bolt/functions/lib/Sequence.cpp:283]: yes
>  https://spark.apache.org/docs/latest/api/sql/index.html#sequence
> Supported types are: byte, short, integer, long, date, timestamp.

**Review by zhangxffff — APPROVED** [review]
LGTM!
